benchmarks/apps/purify/purify.py

```diff
@@ -0,0 +1,136 @@
+import reframe as rfm
+import reframe.utility.sanity as sn
+from benchmarks.modules.utils import SpackTest
+
+class PurifyBase(SpackTest):
+
+    num_tasks = required
+    num_cpus_per_task = required
+
+    valid_systems = ['*']
+    valid_prog_environs = ['default']
+
+    spack_spec = 'purify@4.2.0+benchmarks'
+    executable_opts = ['--benchmark_format=csv',
+                       '--benchmark_out=purify_benchmark.out',
+                       '--benchmark_out_format=csv']
+
+    @sanity_function
+    def validate_solution(self):
+        return sn.assert_found(r'Running', self.stderr)
+
+    @performance_function('ms', perf_key='mean')
+    def extract_mean(self):
+        return sn.extractsingle(r'manual_time_mean\",\S+,(\S+),\S+,ms', self.stdout, 1, float)
+
+    @performance_function('ms', perf_key='median')
+    def extract_median(self):
+        return sn.extractsingle(r'manual_time_median\",\S+,(\S+),\S+,ms', self.stdout, 1, float)
+
+    @performance_function('ms', perf_key='stddev')
+    def extract_stddev(self):
+        return sn.extractsingle(r'manual_time_stddev\",\S+,(\S+),\S+,ms', self.stdout, 1, float)
+
+
+
+@rfm.simple_test
+class PurifyFFTBenchmark(PurifyBase):
+    """
+    Execute all the benchmarks in the fft benchmark on one core with one thread
+    """
+
+    num_tasks = 1
+    num_cpus_per_task = 1
+    env_vars = {
+        'OMP_NUM_THREADS': 1
+    }
```

Comment:
This overrides `env_vars`, it's better to add items to it instead:
```suggestion
    env_vars['OMP_NUM_THREADS'] = 1
```